File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n==1){
            return "1";
        }
        string s=countAndSay(n-1);
        string res;
        int c=0;
        for(int i=0;i<s.length();i++){
            c++;
            if(i==s.length()|| s[i]!=s[i+1]){
                res.append(to_string(c)+s[i]);
                c=0;
            }

        }
        return res;
      


    }
};

File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n==1:
            return "1"
        
        prev=self.countAndSay(n-1)
        c=0
        res=""
        for i in range(len(prev)):
            c+=1
            if i==len(prev)-1 or prev[i]!=prev[i+1]:
                res+=str(c)+prev[i]
                c=0
        return res
        